src/middlewared/middlewared/schema.py

```diff
@@ -467,11 +470,8 @@ def clean(self, value):
                     raise Error(self.name, 'Item#{0} is not valid per list types: {1}'.format(index, found))
         return value
 
-    def dump(self, value):
-        if self.private or (self.items and any(item.private for item in self.items)):
-            return "********"
-
-        return value
+    def has_private(self):
+        return self.private or (self.items and any(item.has_private() for item in self.items))
```

Comment:
`any` will return `False` for empty list
```suggestion
        return self.private or any(item.has_private() for item in self.items)
```